source/kernel/modules.go

```diff
@@ -25,20 +26,39 @@ import (
 const kmodProcfsPath = "/proc/modules"
 
 func getLoadedModules() ([]string, error) {
-	out, err := os.ReadFile(kmodProcfsPath)
-	if err != nil {
-		return nil, fmt.Errorf("failed to read file %s: %s", kmodProcfsPath, err.Error())
-	}
-
-	lines := strings.Split(string(out), "\n")
-	loadedMods := make([]string, len(lines))
-	for _, line := range lines {
-		// skip empty lines
-		if len(line) == 0 {
-			continue
-		}
-		// append loaded module
-		loadedMods = append(loadedMods, strings.Fields(line)[0])
-	}
+  kVersion, err := getVersion()
+  var filesToLoad []string
+  if err != nil {
+    filesToLoad = []string{kmodProcfsPath}
+  } else {
+    kBuiltinModPath := "/lib/modules/" + kVersion + "/modules.builtin"
```

Comment:
We need to specify new `LibDir` in `pkg/utils/hostpath/hostpath.go` and use it here.

And, add corresponding mounts to all deployment files (kustomize and Helm)